src/lib/compress_hw.c

```diff
@@ -119,6 +119,48 @@ static bool _is_codec_type_supported(int fd, struct snd_codec *codec)
 	return found;
 }
 
+static int compress_hw_get_supported_codecs_by_name(const char *name, unsigned int flags, void *codecs, int size)
+{
+	struct snd_compr_caps caps;
+	unsigned int *codecs_buf = (unsigned int *)codecs;
+	unsigned int card, device;
+	unsigned int dev_flag;
+	unsigned int num_codecs, i;
+	int fd, ret;
+	char fn[256];
+
+	if (size < MAX_NUM_CODECS)
+		return oops(&bad_compress, EINVAL, "Input not big enough to hold list of available codecs");
+
+	if (sscanf(&name[3], "%u,%u", &card, &device) != 2)
+		return -EINVAL;
+
+	snprintf(fn, sizeof(fn), "/dev/snd/comprC%uD%u", card, device);
+
+	if (flags & COMPRESS_OUT)
+		dev_flag = O_RDONLY;
+	else
+		dev_flag = O_WRONLY;
+
+	fd = open(fn, dev_flag);
+	if (fd < 0)
+		return oops(&bad_compress, errno, "cannot open device '%s'", fn);
+
+	if (ioctl(fd, SNDRV_COMPRESS_GET_CAPS, &caps)) {
+		close(fd);
+		return oops(&bad_compress, errno, "cannot get device caps");
+	}
+
+	close(fd);
+
+	num_codecs = caps.num_codecs;
```

Comment:
we should check if the num_codecs returned is less than size/(unsigned int)

Reply:
Added the check and another API to get actual count of codecs